src/WelcomeView.vala

```diff
@@ -72,15 +75,16 @@ public class Mail.WelcomeView : Gtk.Box {
         main_box.add (headerbar);
         main_box.add (grid);
 
-        var window_handle = new Hdy.WindowHandle ();
-        window_handle.add (main_box);
+        var window_handle = new Hdy.WindowHandle () {
+            child = main_box
+        };
 
         add (window_handle);
         show_all ();
 
         welcome_button.clicked.connect (() => {
             try {
-                Gtk.show_uri_on_window ((Gtk.Window) get_toplevel (), "settings://accounts/online", Gdk.CURRENT_TIME);
+                AppInfo.launch_default_for_uri ("settings://accounts/online", null);
```

Comment:
We use Gtk.show_uri_on_window here because it includes the event time which fixes issues with focus stealing prevention. `show_uri` still exists in the version of Gtk4 that we have in OS 7, and in future version of Gtk there's `Gtk.UriLauncher.launch`, so I think we should keep this as is for now as it would cause a behavior regression to use AppInfo

Reply:
Should then all AppInfo uses be replaced with `show_uri ` in gtk4 (e.g. the one in the menu)? Or is this a special case?